src/wifi.rs

```diff
@@ -322,6 +340,7 @@ static mut TX_CALLBACK: Option<Box<dyn FnMut(WifiDeviceId, &[u8], bool) + 'stati
 pub struct WifiDriver<'d> {
     status: Arc<mutex::Mutex<(WifiEvent, WifiEvent)>>,
     _subscription: EspSubscription<System>,
+    sysloop: EspSystemEventLoop,
```

Comment:
I feel it is clearer if the `AsyncWifiDriver` itself takes on constructor a `&SystemEventL:oop` or a `SystemEventLoop`, if it needs it, rather than keeping the sysloop of the `WifiDriver` as a state, just so that `AsyncWifiDriver` can re-use it. `SystemEventLoop` is cloneable, and cloning is cheap, as it is just an Arc-ed "()" under the hood, so to say.